File: frontend/src/api/modelConfig.js
// 模型配置 API
import api from './request'

export const modelConfigApi = {
  /**
   * 获取模型配置
   */
  getConfig: () => {
    return api.get('/model-config')
  },

  /**
   * 更新模型配置（完整更新）
   * @param {Object} data - 配置数据
   * @param {string} [data.default_model_type] - 默认模型类型
   * @param {Object} [data.models] - 模型配置
   */
  updateConfig: (data) => {
    return api.put('/model-config', data)
  },

  /**
   * 更新单个模型配置
   * @param {Object} data - 模型数据
   * @param {string} data.model_type - 模型类型 (cloud/local)
   * @param {string} [data.name] - 模型名称
   * @param {string} [data.model_name] - 模型标识
   * @param {string} [data.api_key] - API Key
   * @param {string} [data.base_url] - Base URL
   * @param {string} [data.description] - 描述
   * @param {boolean} [data.enabled] - 是否启用
   */
  updateModel: (data) => {
    return api.patch('/model-config/model', data)
  },

  /**
   * 设置默认模型类型
   * @param {string} modelType - 模型类型 (cloud/local)
   */
  setDefaultType: (modelType) => {
    return api.put(`/model-config/default-type?model_type=${modelType}`)
  }
}

File: backend/app/schemas/model_config.py
"""
模型配置相关 Schema
"""
from typing import Optional
from pydantic import BaseModel


class ModelInfo(BaseModel):
    """单个模型信息"""
    model_name: str
    api_key: str
    base_url: Optional[str] = None
    description: str = ""
    enabled: bool = True


class ModelType(BaseModel):
    """模型类型"""
    type: str
    display_name: str
    description: str = ""


class ModelConfigResponse(BaseModel):
    """模型配置响应"""
    config_name: str
    description: str = ""
    last_updated: str = ""
    default_model_type: str
    models: dict[str, ModelInfo]
    model_types: list[ModelType]


class ModelConfigUpdate(BaseModel):
    """更新模型配置请求"""
    default_model_type: Optional[str] = None
    models: Optional[dict] = None


class ModelInfoUpdate(BaseModel):
    """更新单个模型信息"""
    model_type: str  # cloud 或 local
    model_name: Optional[str] = None
    api_key: Optional[str] = None
    base_url: Optional[str] = None
    description: Optional[str] = None
    enabled: Optional[bool] = None